commonutils.py

```diff
@@ -170,3 +165,45 @@ def setup_nrnbbcore_register_mapping(rings):
                 pc.nrnbbcore_register_mapping(gid, "dend", densec, denseg)
 
     return recordlist
+
+def write_report_config(output_file, report_name, target_name, report_type, report_variable,
+                        unit, report_format, target_type, dt, start_time, end_time, gids,
+                        buffer_size=8):
+    import struct
+    num_gids = len(gids)
+    report_conf = Path(output_file)
+    report_conf.parent.mkdir(parents=True, exist_ok=True)
+    with report_conf.open("wb") as fp:
+        # Write the formatted string to the file
+        fp.write(b"1\n")
+        fp.write(("%s %s %s %s %s %s %d %lf %lf %lf %d %d\n" % (
+            report_name,
+            target_name,
+            report_type,
+            report_variable,
+            unit,
+            report_format,
+            target_type,
+            dt,
+            start_time,
+            end_time,
+            num_gids,
+            buffer_size
+        )).encode())
+        # Write the array of integers to the file in binary format
+        fp.write(struct.pack(f'{num_gids}i', *gids))
+        fp.write(b'\n')
+        fp.write(b"1\n")
```

Comment:
I would maybe have another function to write the spike config (similar to neurodamus).
To facilitate in case in the future we want to add more tests. Something like:
```
def write_spike_config(spike_filename: str, num_population: int, population_names: List[str], population_offsets: List[int]):
```

Reply:
yes, I can add that function to be robust. 
Just note that, 
-  in direct mode, we need to create spike .h5 report explicity in neuron
-  in file mode, we need `coreneuron.skip_write_model_to_disk = True` to be able to create the spike report with coreneuron https://github.com/neuronsimulator/nrn/blob/master/src/coreneuron/io/output_spikes.cpp#L276

The ring tests I added in  https://github.com/neuronsimulator/nrn/pull/2555 produce soma.h5 but not spikes.h5.